### PR TITLE
Add system tests for groups deletion

### PR DESCRIPTION
**Pull request by juliamagan**
|Related issue|
|-------------|
|   #3912           |

## Description

Add a test that checks if groups are deleted correctly.

### Added

- Added `test_remove_group.py`
- Added `cases_remove_group.yaml`
- Added `remove_group_messages.yaml`
- Added `get_group_id()`
---

## Testing performed

| Tester             | Test path | Jenkins | Local   | Commit | Notes                |
|--------------------|-----------|---------|--------|--------|----------------------|
| @juliamagan  (Developer)  |    `tests/system/test_cluster/test_agent_groups/test_remove_group.py`       | :no_entry_sign: :no_entry_sign: :no_entry_sign: | [🟢 ](https://github.com/wazuh/wazuh-qa/files/11221093/R1_test_remove_group.zip)[🟢 ](https://github.com/wazuh/wazuh-qa/files/11221097/R2_test_remove_group.zip)[🟢 ](https://github.com/wazuh/wazuh-qa/files/11221099/R3_test_remove_group.zip) |    91adbe7 | Nothing to highlight |
| @user (Reviewer)   |           | :no_entry_sign: :no_entry_sign: :no_entry_sign: | :no_entry_sign: :no_entry_sign: :no_entry_sign:  |               | Nothing to highlight |

### CODE REVIEW

**Review by Rebits — CHANGES_REQUESTED** [review]
Great job! Some changes are required

File: tests/system/test_cluster/test_agent_groups/test_remove_group.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+    if not metadata['deleted']:
+        HostMonitor(inventory_path=inventory_path,
+                    messages_path=messages_path,
+                    tmp_path=tmp_path).run(update_position=True)
+
+        for manager in test_infra_managers:
+            group_ids[manager] = str(get_group_id(group, manager, host_manager))
+
+    # Check group is deleted or not if exepceted in agent table in every node
```
[the remark this answers is not in the document]
Comment (juliamagan) [reply tests/system/test_cluster/test_agent_groups/test_remove_group.py:108]: Done in a64da1c

File: tests/system/test_cluster/test_agent_groups/test_remove_group.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
+import os
+import pytest
+from time import sleep
+
+import wazuh_testing as fw
```
[the remark this answers is not in the document]
Comment (juliamagan) [reply tests/system/test_cluster/test_agent_groups/test_remove_group.py:5]: Done in 7a8ee99

File: tests/system/test_cluster/test_agent_groups/test_remove_group.py [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+# Variables
+t1_configuration_parameters, t1_configuration_metadata, t1_case_ids = get_test_cases_data(t1_cases_path)
+queries = ['sql select `group` from agent;', 'sql select name from `group`;', 'sql select id_group from belongs;']
+
+
+# Fixtures
+@pytest.fixture()
+def pre_configured_groups(target_node, group):
+    """Fixture to create a group and assign an agent during the setup."""
```
[the remark this answers is not in the document]
Comment (juliamagan) [reply tests/system/test_cluster/test_agent_groups/test_remove_group.py:36]: Done in 4c28ae1

File: tests/system/test_cluster/test_agent_groups/test_remove_group.py [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    # Check that agent has client key file
+    assert check_keys_file(test_infra_agents[0], host_manager), ERR_MSG_CLIENT_KEYS_IN_MASTER_NOT_FOUND
+
+    # Assign agent to group
+    if group != 'default':
+        assign_agent_to_new_group(test_infra_managers[0], group, agent_id, host_manager)
+
+    # Check that agent has group assigned on every node
+    check_agent_groups(agent_id, group, test_infra_managers, host_manager)
+
```
[the remark this answers is not in the document]
Comment (juliamagan) [reply tests/system/test_cluster/test_agent_groups/test_remove_group.py:57]: Done in 79b5ac1

**Review by Rebits — CHANGES_REQUESTED** [review]

**Comment by Rebits** [comment]
### Report :green_circle: 

**Commit**: 04f636f49a79b654c3f3345824794f500e967a36

[report.zip](https://github.com/wazuh/wazuh-qa/files/11147846/report.zip)


> **Note**:
> Only new developed test was tested, it does not ensure correct integration in the system suite

File: tests/system/__init__.py [hunk 5]
```diff
@@ -83,9 +84,18 @@ def assign_agent_to_new_group(host, id_group, id_agent, host_manager):
     host_manager.run_command(host, f"/var/ossec/bin/agent_groups -q -a -i {id_agent} -g {id_group}")
 
 
-def delete_group_of_agents(host, id_group, host_manager):
+def delete_agent_group(host, id_group, host_manager, method='tool'):
     # Delete group
-    host_manager.run_command(host, f"/var/ossec/bin/agent_groups -q -r -g {id_group}")
+    if method == 'tool':
+        host_manager.run_command(host, f"/var/ossec/bin/agent_groups -q -r -g {id_group}")
```
[the remark this answers is not in the document]
Comment (juliamagan) [reply tests/system/__init__.py]: Done in 174821d

File: tests/system/__init__.py [hunk 6]
```diff
@@ -83,9 +84,18 @@ def assign_agent_to_new_group(host, id_group, id_agent, host_manager):
     host_manager.run_command(host, f"/var/ossec/bin/agent_groups -q -a -i {id_agent} -g {id_group}")
 
 
-def delete_group_of_agents(host, id_group, host_manager):
+def delete_agent_group(host, id_group, host_manager, method='tool'):
     # Delete group
```
[the remark this answers is not in the document]
Comment (juliamagan) [reply tests/system/__init__.py]: Done in 91adbe7

**Review by Rebits — APPROVED** [review]
LGTM

**Pull request merged**